website/source/docs/agent/acl-rules.html.md

```diff
@@ -0,0 +1,563 @@
+---
+layout: "docs"
+page_title: "ACL Rules"
+sidebar_current: "docs-agent-acl-rules"
+description: |-
+  Consul provides an optional Access Control List (ACL) system which can be used to control access to data and APIs. The ACL system is a Capability-based system that relies on tokens which can have fine grained rules applied to them. It is very similar to AWS IAM in many ways.
+---
+
+-> **1.4.0 and later:** This guide only applies in Consul versions 1.4.0 and later. The documentation for the legacy ACL system is [here](/docs/guides/acl-legacy.html)
+
+# ACL Rules
+
+Consul provides an optional Access Control List (ACL) system which can be used 
+to control access to data and APIs. To learn more about Consul's ACL review the
+[ACL system documentation](/docs/agent/acl-system.html)
+
+A core part of the ACL system is the rule language, which is used to describe the policy
+that must be enforced. There are two types of rules: prefix based rules and exact matching
+rules. 
+
+## Rule Specification
+
+Rules are composed of a resource, a segment (for some resource areas) and a policy
+disposition. The general structure of a rule is:
+
+```text
+<resource> "<segment>" {
+  policy = "<policy disposition>"
+}
+```
+
+Segmented resource areas allow operators to more finely control access to those resources.
+Note that not all resource areas are segmented such as the `keyring`, `operator`, and `acl` resources. For those rules they would look like:
+
+```text
+<resource> = "<policy disposition>"
+```
+
+Policies can have several control levels:
+
+* `read`: allow the resource to be read but not modified
+* `write`: allow the resource to be read and modified
+* `deny`: do not allow the resource to be read or modified
+
```

Comment:
There is also a "list" disposition but it only applies to the kv store and it controls access to all keys under a segment.

Which this just made me think of the fact that exact match rules should not allow list policies. So the following is valid

```
key_prefix "foo" {
   policy = "list"
}
```

But this is invalid

```
key "foo" {
   policy = "list"
}
```

Reply:
What do you think is the best way to capture this information? A highlighted note or subsection?

Reply:
I would just put it in the policy dispositions lists with a note that it can only be used with the key_prefix resource and if the `acl.enable_key_list_policy` configuration entry is set to true.

Reply:
Updated. LMK if the language needs further modification.